build.go

```diff
@@ -625,18 +639,39 @@ func randString(n int) string {
 	return string(b)
 }
 
-func buildPlatformVolumes(volumes []string) ([]string, error) {
+func buildPlatformVolumes(imgOS string, volumes []string) ([]string, error) {
+	// TODO: Consider a cleaner logic for windows/linux differences
```

Comment:
We should resolve this "TODO". I agree with the statement. 😄 

Suggestion: There are a lot of conditionals throughout this PR. It would be nice it it was abstracted out similar to how it was done for `mountPaths`.